components/org.wso2.carbon.analytics.msf4j.interceptor.common/src/main/java/org/wso2/carbon/analytics/msf4j/interceptor/common/AuthenticationInterceptor.java

```diff
@@ -0,0 +1,161 @@
+    private static final Logger LOGGER = LoggerFactory.getLogger(AuthenticationInterceptor.class);
+
+    @Override
+    public boolean interceptRequest(Request request, Response response) throws Exception {
+        if (!DataHolder.getInstance().isInterceptorEnabled()) {
+            return true;
+        } else {
+            for (String url : DataHolder.getInstance().getExcludeURLList()) {
+                if (request.getUri().matches(url)) {
+                    return true;
+                }
+            }
+            IdPClient idPClient = DataHolder.getInstance().getIdPClient();
+            Headers headers = request.getHeaders();
+            String authorizationHeader = request.getHeader(IdPClientConstants.AUTHORIZATION_HEADER);
+            if (authorizationHeader != null) {
+                String headerPrefix = authorizationHeader.split(" ")[0];
+                String headerPostfix = authorizationHeader.split(" ")[1];
+                if (headerPostfix != null) {
+                    if (headerPrefix.equalsIgnoreCase(IdPClientConstants.BEARER_PREFIX)) {
+                        String cookieHeader = headers.get(IdPClientConstants.COOKIE_HEADER);
+                        String partialTokenFromCookie = null;
+                        if (cookieHeader != null) {
+                            cookieHeader = cookieHeader.trim();
+                            String[] cookies = cookieHeader.split(";");
+                            String token2 = Arrays.stream(cookies)
+                                    .filter(name -> name.contains(IdPClientConstants.WSO2_SP_TOKEN_2))
+                                    .findFirst().orElse("");
+                            String tokensArr[] = token2.split("=");
+                            if (tokensArr.length == 2) {
+                                partialTokenFromCookie = tokensArr[1];
+                            }
+                        }
+                        String accessToken = (partialTokenFromCookie != null) ?
+                                headerPostfix + partialTokenFromCookie :
+                                headerPostfix;
+                        String username = idPClient.authenticate(accessToken);
+                        if (username != null) {
+                            request.setProperty(InterceptorConstants.PROPERTY_USERNAME, username);
+                            return true;
+                        }
+                        return false;
+                    } else if (headerPrefix.equalsIgnoreCase(IdPClientConstants.BASIC_PREFIX)) {
+                        byte[] decodedAuthHeader = Base64.getDecoder().decode(headerPostfix);
+                        String authHeader = new String(decodedAuthHeader, Charset.forName("UTF-8"));
+                        String userName = authHeader.split(":")[0];
+                        String password = authHeader.split(":")[1];
+                        String appName = request.getUri().split("/\\|?")[1];
```

Comment:
I believe what you are trying to get here as the `appName` is the part of the URI between the first `/` and the second `/`, right?
If so, you can find the position of the second `/` with `String.indexOf(int ch, int fromIndex))` and get a substring.